### DISCUSSION TITLE
Problems Installing sdmTMB on server

### DISCUSSION THREAD

**Discussion by sivasquezp** [sdmTMB/sdmTMB#226] [Q&A] [answered]
Hello,

I have used sdmTMB recently but since this week I have tried to use it on a server and I have not been able to install it probably because of the old operating system: CentOS Stream-8.

I had the same problem with INLA before, but I fixed it with the help of Håvard Rue through the option:
inla.binary.install()

Is there any similar alternative for sdmTMB?

Thanks in advance!

Sebastián

**Comment by seananderson** [comment]
Are you installing from source or binary and from CRAN or GitHub? If the
CRAN binary isn’t built, does building from source work? If you’re
installing from source, you should probably just install the GitHub
version. We’ll be pushing an update soon.

**Reply by sivasquezp** [reply]
Thanks Sean!

I have tried from CRAN using:
install.packages("sdmTMB", dependencies = TRUE)
I also tried from Github using:
install.packages("remotes")
remotes::install_github("pbs-assess/sdmTMB", dependencies = TRUE)
Finally I downloaded the sdmTMB_0.3.0.tar.gz binary and tried to install it using:
install.packages("sdmTMB_0.3.0.tar.gz", repos=NULL)

All attempts were unsuccessful, with errors probably associated with missing libraries in my version of Lynux:

![image](https://github.com/pbs-assess/sdmTMB/assets/133794441/834a91ff-e8d1-40f2-ba51-d64966d25441)

**Reply by seananderson** [reply, marked as answer]
I would start by getting nloptr to install.
